plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts

```diff
@@ -9,6 +9,16 @@ import { type AttributeMetadata } from '@woocommerce/types';
  */
 import { WooCommerceBlockLocation } from '../product-template/utils';
 
+export enum ProductCollectionUIStatesInEditor {
+	PRODUCT_CONTEXT_PICKER = 'product_context_picker',
+	USES_REFERENCE_PREVIEW_MODE = 'uses_reference_preview_mode',
+	COLLECTION_CHOOSER = 'collection_chooser',
+	VALID = 'valid',
+	// Future states
+	// ERROR_STATE = 'error_state',
+	// DELETED_PRODUCT_STATE = 'deleted_product_state',
```

Comment:
I'd unify them a bit:

```suggestion
	COLLECTION_PICKER = 'collection_picker,
	PRODUCT_PICKER = 'product_picker', // or with REFERENCE as per Chris suggestion
	VALID_WITH_PREVIEW = 'valid_with_preview',
	VALID = 'valid',
```

Those are just examples and don't have to be use but you know what I mean:
- chooser vs picker - we can use one of them
- valid vs uses_reference_preview_mode - these are essentially the same but one adds preview label. I think we can make it more similar.

Reply:
> chooser vs picker - we can use one of them
> 

Makes sense. I will make the changes 🤝🏻

> valid vs uses_reference_preview_mode - these are essentially the same but one adds preview label. I think we can make it more similar.
> 

I was thinking about going with this but then I decided to go with `uses_reference_preview_mode`. Only reason behind this was that there is [another logic](https://github.com/woocommerce/woocommerce/blob/67fa07f2a8bb932cb793828d914549072b7e7959/plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx#L327-L364) which decide if Product Collection should be in preview mode or not. Therefore, I wanted to avoid that confusion using a better name.

Probably that logic can be moved to [getProductCollectionUIStateInEditor](https://github.com/woocommerce/woocommerce/blob/67fa07f2a8bb932cb793828d914549072b7e7959/plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx#L194). I wanted to avoid this because that will require some testing to make sure we are not breaking anything.

@kmanijak Should we consider moving that logic as part of this PR, or can we create an issue to handle that later to avoid scope creep in this PR?

In any case, I will change the name to `VALID_WITH_PREVIEW`.

Here is how final enum will look like:

```tsx
export enum ProductCollectionUIStatesInEditor {
	COLLECTION_PICKER = 'collection_picker,
	PRODUCT_REFERENCE_PICKER = 'product_reference_picker',  
        VALID = 'valid',  
	VALID_WITH_PREVIEW = 'valid_with_preview',
        // Future states  
       // INVALID = 'invalid',  
       // DELETED_PRODUCT_REFERENCE = 'deleted_product_reference',
}
```

CC: @xristos3490

Reply:
[be8d125](https://github.com/woocommerce/woocommerce/pull/50164/commits/be8d1259631c9c80fd67ced80de7bfb4eaff4169)